src/main/java/com/google/cloud/tools/app/impl/cloudsdk/internal/process/DefaultProcessRunner.java

```diff
@@ -19,34 +19,36 @@
 import java.util.List;
 import java.util.Map;
 import java.util.Scanner;
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.TimeUnit;
+import java.util.logging.Logger;
 
 /**
  * Default process runner that allows synchronous or asynchronous execution. It also allows
  * monitoring output and checking the exit code of the child process.
  */
 public class DefaultProcessRunner implements ProcessRunner {
+  private static final Logger log = Logger.getLogger(DefaultProcessRunner.class.toString());
 
-  private ProcessBuilder processBuilder;
-  private boolean async = false;
+  private boolean async;
   private ProcessOutputLineListener stdOutLineListener;
   private ProcessOutputLineListener stdErrLineListener;
   private ProcessExitListener exitListener;
+  private String waitSuccessMessage;
+  private int waitSuccessTimeoutSeconds;
 
   private Process process;
+  private CountDownLatch waitSuccessLatch;
 
-  /**
-   * Create the process runner with a default process builder, with inheritIO enabled.
-   */
-  public DefaultProcessRunner() {
-    this(new ProcessBuilder());
-    processBuilder.inheritIO();
-  }
+  private Map<String, String> environment;
 
-  /**
-   * Create the process runner with the provided process builder.
-   */
-  public DefaultProcessRunner(ProcessBuilder processBuilder) {
-    this.processBuilder = processBuilder;
+  private DefaultProcessRunner(Builder builder) {
+    this.async = builder.async;
+    this.stdOutLineListener = builder.stdOutLineListener;
+    this.stdErrLineListener = builder.stdErrLineListener;
+    this.exitListener = builder.exitListener;
+    this.waitSuccessMessage = builder.waitSuccessMessage;
```

Comment:
I think, ultimately, we might want to support regular expressions for successMessage, but otherwise, I think it's pretty generic for any process execution.